spec.html

```diff
@@ -772,7 +772,7 @@ <h1>Grammar Notation</h1>
   <emu-clause id="sec-algorithm-conventions">
     <h1>Algorithm Conventions</h1>
     <p>The specification often uses a numbered list to specify steps in an algorithm. These algorithms are used to precisely specify the required semantics of ECMAScript language constructs. The algorithms are not intended to imply the use of any specific implementation technique. In practice, there may be more efficient algorithms available to implement a given feature.</p>
-    <p>Algorithms may be explicitly parameterized, in which case the names and usage of the parameters must be provided as part of the algorithm's definition.</p>
+    <p>Algorithms may be explicitly parameterized, in which case the names and usage of the parameters must be provided as part of the algorithm's definition. <dfn>Optional parameters</dfn> are denoted with surrounding brackets (`[ , _name_ ]`) and are no different from mandatory parameters within algorithm steps. A <dfn>rest parameter</dfn> may appear at the end of a parameter list, denoted with leading elipsis (`...name`). The rest parameter captures all of the arguments provided following the mandatory and optional parameters, if any, into a possibly-empty List.</p>
```

Comment:
Is this section meant to apply to both built-ins and to AOs and ACs? We need to say something about the mandatory nature of parameters, but it's treated differently from built-ins to AOs. I assume it's an error for AOs to use a parameter with an absent argument, but built-ins can and assume it's undefined, right?

Reply:
The handling of missing mandatory parameters is detailed in [section 17](https://tc39.es/ecma262/#sec-ecmascript-standard-built-in-objects). The paragraph here is written in a way that applies equally to built-ins and AOs.

Reply:
> We need to say something about the mandatory nature of parameters, but it's treated differently from built-ins to AOs. I assume it's an error for AOs to use a parameter with an absent argument, but built-ins can and assume it's undefined, right?

I don't think we need anything further: the behavior for built-ins is already spelled out below, in [Clause 17](https://tc39.es/ecma262/#sec-ecmascript-standard-built-in-objects):

> Unless otherwise specified in the description of a particular function, if a built-in function or constructor is given fewer arguments than the function is specified to require, the function or constructor shall behave exactly as if it had been given sufficient additional arguments, each such argument being the undefined value. Such missing arguments are considered to be “not present” and may be identified in that manner by specification algorithms.

And of course for AOs and ACs it would be a _spec_ error; we don't need to spell out the behavior for invoking an AO or AC without all required arguments because the situation does not arise.

(It's not entirely clear to me why the description of editorial conventions for built-in functions is split across [Clause 9.3](https://tc39.es/ecma262/#sec-built-in-function-objects) "Built-In Function Objects" and Clause 17 "ECMAScript Standard Built-in Objects"; that could perhaps be improved in a different PR.)

Reply:
> (It's not entirely clear to me why the description of editorial conventions for built-in functions is split across [Clause 9.3](https://tc39.es/ecma262/#sec-built-in-function-objects) "Built-In Function Objects" and Clause 17 "ECMAScript Standard Built-in Objects"; that could perhaps be improved in a different PR.)

See PR #539.

Reply:
@syg Do you think the parts I've mentioned are sufficient, or would you still like something more here?

Reply:
They seem sufficient to me.